src/bonds.h

```diff
@@ -119,19 +149,21 @@ struct PeriodicDihedral : public BondData {
     void setEnergyFunction(const ParticleVector &p);
     PeriodicDihedral() = default;
     PeriodicDihedral(double k, double phi, double n, const std::vector<int> &index);
-}; // end of PeriodicDihedral
+};
 
 /*
  * Serialize to/from json
  */
 
-void to_json(json &j, const std::shared_ptr<BondData> &b);
 void from_json(const json &j, std::shared_ptr<BondData> &b);
+void to_json(json &j, const std::shared_ptr<const BondData> &b);
+void to_json(Faunus::json &j, const BondData &b);
 
 void setBondEnergyFunction(std::shared_ptr<BondData> &b,
                            const ParticleVector &p); //!< Set the bond energy function of `BondData` which
                                                      //!< require a reference to the particle vector
 
+[[deprecated("Use bonds.find<TClass>() method instead.")]]
 inline auto filterBonds(const std::vector<std::shared_ptr<BondData>> &bonds, BondData::Variant bondtype) {
     std::vector<std::shared_ptr<BondData>> filt;
```

Comment:
I forgot why BondData needs to be stored as pointers? This could lead to excessive memory fragmentation compared to a straight vector.

Reply:
`BondData` is an abstract class for all bonded interactions hence pointers. In general I do not worry so much if the data structure is accessed only outside the MC loop.